pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/LoaderUtils.java

```diff
@@ -90,4 +97,56 @@ public static void writeIndexToV3Format(SegmentDirectory.Writer segmentWriter, S
     }
     return stringList;
   }
+
+  /**
+   * Try to recover a segment from reload failure (reloadSegment() method in HelixInstanceDataManager), no effect for
```

Comment:
For normal exceptions, we also use this method to recover segments (this method is called inside reloadSegment() finally block). Updated the comments.